File: engines/austin/contracts/engine/EngineRegistry.ts
/**
 * ============================================================
 * ENGINE REGISTRY CONTRACT
 * ============================================================
 */

import { EngineContract } from "./EngineContract";

export interface EngineRegistry {

    register(engine: EngineContract): Promise<void>;

    unregister(id: string): Promise<void>;

    get(id: string): EngineContract | undefined;

    getAll(): EngineContract[];

    exists(id: string): boolean;

    clear(): Promise<void>;

}

File: app/austin/runtime/router/engine_registry.py
"""
Austin Engine Registry

Maintains the collection of
available runtime engines.
"""


class EngineRegistry:

    def __init__(self):
        self._engines = {}

    def register(
        self,
        name,
        engine,
    ):
        self._engines[name] = engine

    def get(
        self,
        name,
    ):
        return self._engines.get(name)

    def exists(
        self,
        name,
    ):
        return name in self._engines

    def names(self):
        return sorted(self._engines.keys())

    def summary(self):
        return {
            "engines": len(self._engines),
            "names": self.names(),
        }